ospf6d/ospf6_interface.h

```diff
@@ -95,6 +115,9 @@ struct ospf6_interface {
 	/* MTU mismatch check */
 	uint8_t mtu_ignore;
 
+	/* Authentication trailer related config */
+	struct ospf6_auth_data at_data;
```

Comment:
Next problem:  the sequence number is not counted on the interface, it is counted on the entire VRF / instance.  RFC7166 says:

> The 64-bit sequence number MUST be incremented for every OSPFv3 packet sent by the OSPFv3 router.

"Every OSPFv3 packet" - there is nothing about interface, neighbor, etc. here. 

(This actually makes things easier, especially for saving the persistent seqnum, since we only need to save one seqnum per VRF / instance.)

Reply:
@eqvinox: I think we need to increment the sequence number for every packet that is sent out. RFC does not demand how we implement it.
As you mentioned you can implement it at the router level, but it can be implemented at the interface level also. 
Ya its easy to store a single sequence number for the whole process but better to have sequence number at interface level.
When we store the sequence number we need to store the higher sequence number. 
and when the router restarts we need to increment the higher sequence number. 

Now the problem is how we store the sequence number into a persistent storage?
Apart from storing it into a file. do you have any suggestion ?

And as per your last comment do we need to have a different sequence number based on different ospf packet type. It will increase complexity of storing the sequence number?

Reply:
I didn't understand this: "better to have sequence number at interface level". What does that mean - how could it be "better"?

Reply:
If you keep sequence number at interface level it will delay the sequence number wrap scenario.
And it's already implemented at interface level so better keep it as is.
regarding storage of sequence number in persistent memory we can store it for the biggest higher sequence number among all neighbors and after restart increase that and copy it to all interface.

Reply:
> RFC does not demand how we implement it.

It does.

```
   o  Cryptographic Sequence Number

      This is a 64-bit strictly increasing sequence number that is used
      to guard against replay attacks.  The 64-bit sequence number MUST
      be incremented for every OSPFv3 packet sent by the OSPFv3 router.
```

"incremented for every OSPFv3 packet"

The counter cannot be per interface.  Otherwise you can do a replay attack with packets from one interface/link by sending them on another interface/link.

> but better to have sequence number at interface level.

**NO.  It is strictly worse.**